src/bun.js/test/jest.zig

```diff
@@ -430,6 +493,213 @@ pub const TestRunner = struct {
 pub const Jest = struct {
     pub var runner: ?*TestRunner = null;
 
+    pub const Snapshots = struct {
+        pub var snapshot_count: usize = 0;
+        pub var update_snapshots: bool = false;
+        pub var initialized: bool = false;
+        pub var file_header = "// Bun Snapshot v1, https://goo.gl/fbAQLP\n";
+
+        pub var snapshot_dir: std.fs.Dir = undefined;
+        pub var snapshot_dir_path: []const u8 = undefined;
+        pub var snapshot_file: std.fs.File = undefined;
+
+        pub var file_buf: std.ArrayList(u8) = undefined;
+        pub const ValuesHashMap = std.HashMap(usize, string, bun.IdentityContext(usize), std.hash_map.default_max_load_percentage);
+        pub var values: ValuesHashMap = undefined;
+        pub var counts: bun.StringHashMap(usize) = undefined;
+
+        pub var current_file_id: TestRunner.File.ID = undefined;
+
+        pub fn getOrPut(expect: *Expect, value: JSValue, hint: string, globalObject: *JSC.JSGlobalObject) !?string {
+            const snapshot_name = try expect.getSnapshotName(default_allocator, hint);
+
+            if (current_file_id != expect.scope.file_id or !initialized) {
+                current_file_id = expect.scope.file_id;
+
+                try cleanup();
+
+                counts = bun.StringHashMap(usize).init(default_allocator);
+                values = ValuesHashMap.init(default_allocator);
+
+                var dir_path = runner.?.files.get(current_file_id).source.path.name.dirWithTrailingSlash();
+                if (!initialized) snapshot_dir = try getSnapshotDir(dir_path);
```

Comment:
`getSnapshotDir` seems likely to be buggy

How does Jest determine where to put it? Is it the nearest package.json?

Reply:
the snapshot dir is placed in the same directory as the test file with `toMatchSnapshot()`